tasks.c

```diff
@@ -1050,33 +1051,33 @@ static void prvAddNewTaskToReadyList( TCB_t * pxNewTCB ) PRIVILEGED_FUNCTION;
 
                     uxCoreMap &= ( ( 1 << configNUM_CORES ) - 1 );
 
-                    while( uxCoreMap != 0 )
+                    for( x = ( configNUM_CORES - 1 ); x >= 0; x-- )

```

Comment:
Indeed, the following is the possible value of UBaseType_t in FreeRTOS porting.
```
typedef portUBASE_TYPE UBaseType_t;
typedef uint32_t UBaseType_t;
typedef uint64_t UBaseType_t;
typedef uint8_t UBaseType_t;
typedef unsigned char UBaseType_t;
typedef unsigned long UBaseType_t;
typedef unsigned portBASE_TYPE UBaseType_t;
typedef unsigned short UBaseType_t;
```
We should use sizeof( UBaseType_t ) * CHAR_BIT to check. However, sizeof can't be used in preprocessor condition. The UBaseType_t size is decided by porting.
Suggest we remove the check here.